packages/kbn-router-to-openapispec/src/oas_converter/zod/lib.ts

```diff
@@ -0,0 +1,215 @@
+    }
+  }
+  return type;
+};
+
+interface NativeEnumType {
+  [k: string]: string | number;
+  [nu: number]: string;
+}
+
+type ZodTypeLikeString =
+  | z.ZodString
+  | z.ZodOptional<ZodTypeLikeString>
+  | z.ZodDefault<ZodTypeLikeString>
+  | z.ZodEffects<ZodTypeLikeString, unknown, unknown>
+  | z.ZodUnion<[ZodTypeLikeString, ...ZodTypeLikeString[]]>
+  | z.ZodIntersection<ZodTypeLikeString, ZodTypeLikeString>
+  | z.ZodLazy<ZodTypeLikeString>
+  | z.ZodLiteral<string>
+  | z.ZodEnum<[string, ...string[]]>
+  | z.ZodNativeEnum<NativeEnumType>;
+
+const instanceofZodTypeLikeString = (_type: z.ZodTypeAny): _type is ZodTypeLikeString => {
+  const type = unwrapZodType(_type, false);
+
+  if (instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodEffects)) {
+    if (type._def.effect.type === 'preprocess') {
+      return true;
+    }
+  }
+  if (instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodUnion)) {
+    return !type._def.options.some((option) => !instanceofZodTypeLikeString(option));
+  }
+  if (instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodArray)) {
+    return instanceofZodTypeLikeString(type._def.type);
+  }
+  if (instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodIntersection)) {
+    return (
+      instanceofZodTypeLikeString(type._def.left) && instanceofZodTypeLikeString(type._def.right)
+    );
+  }
+  if (instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodLiteral)) {
+    return typeof type._def.value === 'string';
+  }
+  if (instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodEnum)) {
+    return true;
+  }
+  if (instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodNativeEnum)) {
+    return !Object.values(type._def.values).some((value) => typeof value === 'number');
+  }
+  return instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodString);
+};
+
+const zodSupportsCoerce = 'coerce' in z;
+
+type ZodTypeCoercible = z.ZodNumber | z.ZodBoolean | z.ZodBigInt | z.ZodDate;
+
+const instanceofZodTypeCoercible = (_type: z.ZodTypeAny): _type is ZodTypeCoercible => {
+  const type = unwrapZodType(_type, false);
+  return (
+    instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodNumber) ||
+    instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodBoolean) ||
+    instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodBigInt) ||
+    instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodDate)
+  );
+};
+
+const convertObjectMembersToParameterObjects = (
+  shape: z.ZodRawShape,
+  isRequired: boolean,
+  isPathParameter = false,
+  knownParameters: KnownParameters = {}
+): OpenAPIV3.ParameterObject[] => {
+  return Object.entries(shape).map(([shapeKey, subShape]) => {
+    const isSubShapeRequired = !subShape.isOptional();
+
+    if (!instanceofZodTypeLikeString(subShape)) {
+      if (zodSupportsCoerce) {
+        if (!instanceofZodTypeCoercible(subShape)) {
+          throw createError(
+            `Input parser key: "${shapeKey}" must be ZodString, ZodNumber, ZodBoolean, ZodBigInt or ZodDate`
```

Comment:
In the `instanceofZodTypeCoercible` (above) you don't have a condition for `ZodString`. Is that normal?